src/python/grpcio/grpc/_common.py

```diff
@@ -97,3 +98,27 @@ def serialize(message, serializer):
 def deserialize(serialized_message, deserializer):
   return _transform(serialized_message, deserializer,
                     'Exception deserializing message!')
+
+
+class CleanupThread(threading.Thread):
+  """A wrapper around thread for custom join behavior.
+  
+  On Interpreter exit, Python will attempt to join outstanding threads
+  prior to garbage collection.  We may need to do additional cleanup, and
+  we acomplish this by wrapping the join() method.
+  """
+  
+  def __init__(self, group=None, target=None, join_target=None, name=None,
+               args=(), join_args=(), kwargs={}, join_kwargs={}):
+    """join_target() will be called with a timeout,
+    followed by join_args then join_kwargs."""
+    super(CleanupThread, self).__init__(group=group, target=target, 
+                                     name=name, args=args, kwargs=kwargs)
+    self._join_target = join_target
+    self._join_args = join_args
+    self._join_kwargs = join_kwargs
+
+  def join(self, timeout=None):
+    if self._join_target:
```

Comment:
gRPC Python style is to always use `is None` and `is not None` when comparing against the `None` object.